From 5d57983ce9d566e1e27862ecd4754d6a343523a3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:35 -0400
Subject: [usb] storage/ene_ub6250: Use kmemdup instead of kmalloc + memcpy

Message-id: <1410887463-20674-504-git-send-email-dzickus@redhat.com>
Patchwork-id: 95648
O-Subject: [RHEL7 PATCH 503/591] USB: storage: ene_ub6250: Use kmemdup instead of kmalloc + memcpy
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a328512d3cb9e7b5d998eeb2675216edf4407f77
Author: Benoit Taine <benoit.taine@lip6.fr>
Date:   Mon May 26 17:21:10 2014 +0200

    USB: storage: ene_ub6250: Use kmemdup instead of kmalloc + memcpy

    This issue was reported by coccicheck using the semantic patch
    at scripts/coccinelle/api/memdup.cocci

    Signed-off-by: Benoit Taine <benoit.taine@lip6.fr>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/ene_ub6250.c b/drivers/usb/storage/ene_ub6250.c
index 1bfc9a6..ef6efb5 100644
--- a/drivers/usb/storage/ene_ub6250.c
+++ b/drivers/usb/storage/ene_ub6250.c
@@ -1928,11 +1928,10 @@ static int ene_load_bincode(struct us_data *us, unsigned char flag)
   usb_stor_dbg(us, "load firmware %s failed\n", fw_name);
   goto nofw;
  }
- buf = kmalloc(sd_fw->size, GFP_KERNEL);
+ buf = kmemdup(sd_fw->data, sd_fw->size, GFP_KERNEL);
  if (buf == NULL)
   goto nofw;
 
- memcpy(buf, sd_fw->data, sd_fw->size);
  memset(bcb, 0, sizeof(struct bulk_cb_wrap));
  bcb->Signature = cpu_to_le32(US_BULK_CB_SIGN);
  bcb->DataTransferLength = sd_fw->size;
-- 
1.7.1